PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-azure-sdk/src/main/java/com/microsoft/azure/toolkit/intellij/azure/sdk/buildtool/UseOfBlockOnAsyncClientsCheck.java

```diff
@@ -0,0 +1,165 @@
+ */
+public class UseOfBlockOnAsyncClientsCheck extends LocalInspectionTool {
+
+    /**
+     * This method is used to get the visitor for the inspection tool.
+     * The visitor is used to check for the use of block() method on async clients in Azure SDK.
+     * The visitor checks if the method call is a block() method call on a reactive type
+     * and if the reactive type is an async client in Azure SDK.
+     * If the method call is a block() method call on an async client, it reports a problem.
+     */
+    @NotNull
+    @Override
+    public JavaElementVisitor buildVisitor(@NotNull ProblemsHolder holder, boolean isOnTheFly) {
+        return new UseOfBlockOnAsyncClientsVisitor(holder);
+    }
+
+    /**
+     * Visitor to check for the use of block() method on async clients in Azure SDK.
+     * The visitor checks if the method call is a block() method call on a reactive type
+     * and if the reactive type is an async client in Azure SDK.
+     * If the method call is a block() method call on an async client, it reports a problem.
+     */
+    static class UseOfBlockOnAsyncClientsVisitor extends JavaElementVisitor {
+
+        private final ProblemsHolder holder;
+
+        // Define constants for string literals
+        private static final RuleConfig RULE_CONFIG;
+        private static final boolean SKIP_WHOLE_RULE;
+
+        static {
+            final String ruleName = "UseOfBlockOnAsyncClientsCheck";
+            RuleConfigLoader centralRuleConfigLoader = RuleConfigLoader.getInstance();
+
+            // Get the RuleConfig object for the rule
+            RULE_CONFIG = centralRuleConfigLoader.getRuleConfig(ruleName);
+            SKIP_WHOLE_RULE = RULE_CONFIG.skipRuleCheck() || RULE_CONFIG.getListedItemsToCheck().isEmpty();
+        }
+
+        /**
+         * Constructor to initialize the visitor with the holder
+         *
+         * @param holder ProblemsHolder - the holder to report the problems
+         */
+        UseOfBlockOnAsyncClientsVisitor(ProblemsHolder holder) {
+            this.holder = holder;
+        }
+
+        /**
+         * This method is used to visit the method call expression.
+         * The method call expression is checked to see if it is a block() method call on an async client.
+         * If the method call is a block() method call on an async client, it reports a problem.
+         *
+         * @param expression PsiMethodCallExpression - the method call expression to visit
+         */
+        @Override
+        public void visitMethodCallExpression(@NotNull PsiMethodCallExpression expression) {
+            super.visitMethodCallExpression(expression);
+
+            if (SKIP_WHOLE_RULE) {
+                return;
+            }
+
+            // Check if the method call is block() or a variant of block() method
+            if (expression.getMethodExpression().getReferenceName().startsWith(RULE_CONFIG.getMethodsToCheck().get(0))) {
```

Comment:
Why do we check only for the first method `get(0)`? Should we check for all the methods listed in the rule config?

Reply:
I expanded this to have multiple blocking methods so it's no longer `.get(0)`

the `.get(0)` is for json entries that are Strings, but there's other similar json entries that are Lists, so to avoid having a duplicate `List methodToCheck` and `String methodToCheck`, I return a list which can either have a single entry or multiple